pkg/endpoint/policy.go

```diff
@@ -1107,14 +1107,16 @@ func (e *Endpoint) SetIdentity(identity *identityPkg.Identity, newEndpoint bool)
 	}
 
 	// Current security identity for endpoint is its old identity - delete its
-	// reference from global identity manager, add add a reference to the new
-	// identity for the endpoint.
-	if newEndpoint {
+	// reference from global identity manager, add a reference to the new
+	// identity for the endpoint. If the endpoint has never had its identitySet,
+	// we should note remove the old identity from the identity manager
+	if newEndpoint || !e.identitySet {
```

Comment:
Been thinking a bit here, and I think the correct behavior here is;
```suggestion
	if !e.identitySet {
```

This in case the restore is slow and it ends up changing identity in another code path, resulting in it never freeing it. I have to repro to check if that can actually happen, but I have a hunch that it is indeed possible..